### ISSUE TITLE
Walk movement: moving platforms

### ISSUE THREAD

**Issue by Brungleby** [Brungleby/Mithril.Modules#8]
Currently this works fine on the vertical axis but the platform slides out underneath the player laterally. This may be because the player collider currently has no friction per its physical material.

**Comment by Brungleby** [comment]
I have discovered firstly that this happened because I was using `transform.position = ()` on the platform. Instead, one should use `rigidbody.MovePosition()` to allow this for normal physics interactions.

**Comment by Brungleby** [comment]
I've also discovered that the issue is isolated for the player for a number of factors combined: the friction component, the physical material, and the loss of gravity while grounded all prevent the pawn from moving along the platform.

Removing these functionalities and using "normal" physics achieves the desired result and an overall more "physically accurate" feel, but sprouts new issues that will probably require an overhaul of the walk component to resolve.

Therefore I have decided now to keep the existing setup and develop additional code to patch the issue like normal. The reason for this is to maintain a certain level of control over the pawn's movement and allow for a movement style that feels both "physical" and "simple".
